### PR TITLE
Enable fuzzer

### PR DESCRIPTION
**Pull request by atobiszei**
How to build:
FUZZER_BUILD=1 CHECK_COVERAGE=0 RUN_TESTS=0 BASE_OS=ubuntu OVMS_CPP_DOCKER_IMAGE=u_ovms_fuzz make targz_package
Fuzzer build won't run test at this point.

* Fix few gcc warnings issues

### CODE REVIEW

File: Dockerfile.ubuntu [hunk 1]
```diff
@@ -305,9 +305,8 @@ RUN if [ "$ov_use_binary" == "1" ] ; then true ; else exit 0 ; fi ; sed -i -e "s
 RUN if [ "$ov_use_binary" == "0" ] ; then true ; else exit 0 ; fi ; sed -i -e "s#REPLACE_OPENVINO_NAME#$(git --git-dir /openvino/.git log -n 1 | head -n 1 | cut -d' ' -f2 | head -c 12)#g" /ovms/src/version.hpp
 
 ARG RUN_TESTS=1
-ENV MEDIAPIPE_DISABLE_PARAMS=" --define MEDIAPIPE_DISABLE=1 --cxxopt=-DMEDIAPIPE_DISABLE=1 "
 # Test mediapipe build
-RUN  if [ "$RUN_TESTS" == "1" ] ; then bazel test ${debug_bazel_flags}${MEDIAPIPE_DISABLE_PARAMS} --jobs=$JOBS --test_summary=detailed --test_output=streamed //src:ovms_test > ${TEST_LOG} 2>&1 || (cat ${TEST_LOG} && rm -rf ${TEST_LOG} && exit 1 ; ) && tail -n 100 ${TEST_LOG} && rm -rf ${TEST_LOG} ; fi ;
+RUN  if [ "$RUN_TESTS" == "1" ] ; then bazel test ${debug_bazel_flags} --jobs=$JOBS --test_summary=detailed --test_output=streamed //src:ovms_test > ${TEST_LOG} 2>&1 || (cat ${TEST_LOG} && rm -rf ${TEST_LOG} && exit 1 ; ) && tail -n 100 ${TEST_LOG} && rm -rf ${TEST_LOG} ; fi ;
```
Comment (rasapala) [inline Dockerfile.ubuntu:309]: Remove this whole line.

File: src/BUILD [hunk 2]
```diff
@@ -397,7 +410,16 @@ cc_library(
         "-Wall",
         "-Wno-unknown-pragmas",
         "-Werror",
-    ],
+        ] + select({
+            "//conditions:default": [
+            ],
+            "//src:fuzzer_build" : [
+                "-fsanitize=address,undefined,",
```
Comment (ngrozae) [inline src/BUILD:417]: address sanitizer and ub sanitizer can't be enabled at the same time
```suggestion
                "-fsanitize=address",
```

File: src/BUILD [hunk 3]
```diff
@@ -397,7 +410,16 @@ cc_library(
...
+                "-fprofile-generate",
+                "-ftest-coverage",
+                "-fprofile-arcs",
```
Comment (ngrozae) [inline src/BUILD:420]: do we need -fprofile-arcs if -fprofile-generate is used?

File: src/BUILD [hunk 4]
```diff
@@ -406,7 +428,18 @@ cc_library(
...
-    ],
+    ] + select({
+        "//conditions:default": [
+        ],
+        "//src:fuzzer_build" : [
+            "-fprofile-generate",
+            "-fprofile-arcs",
+            "-fsanitize=address",
+            "-fsanitize-coverage=trace-pc",
+            "-fsanitize-coverage=trace-pc-guard",
```
Comment (ngrozae) [inline src/BUILD:439]: is that option supported for gcc?

File: src/BUILD [hunk 5]
```diff
@@ -406,7 +428,18 @@ cc_library(
         "-luuid",
         "-lstdc++fs",
         "-lcrypto",
-    ],
+    ] + select({
+        "//conditions:default": [
+        ],
+        "//src:fuzzer_build" : [
+            "-fprofile-generate",
+            "-fprofile-arcs",
```
Comment (ngrozae) [inline src/BUILD:436]: as above do we need -fprofile-arcs if -fprofile-generate is used?

**Review by ngrozae — APPROVED** [review]

**Review by rasapala — APPROVED** [review]
LGTM

**Pull request merged**